packages/scm/src/browser/scm-widget.tsx

```diff
@@ -17,82 +17,66 @@
+export class ScmWidget extends BaseWidget implements StatefulWidget {
 
-    static ID = 'scm-view';
-
-    static RESOURCE_GROUP_CONTEXT_MENU = ['RESOURCE_GROUP_CONTEXT_MENU'];
-    static RESOURCE_GROUP_INLINE_MENU = ['RESOURCE_GROUP_INLINE_MENU'];
+    protected panel: Panel;
 
-    static RESOURCE_INLINE_MENU = ['RESOURCE_INLINE_MENU'];
-    static RESOURCE_CONTEXT_MENU = ['RESOURCE_CONTEXT_MENU'];
+    static ID = 'scm-view';
 
-    @inject(CorePreferences) protected readonly corePreferences: CorePreferences;
     @inject(ScmService) protected readonly scmService: ScmService;
-    @inject(CommandRegistry) protected readonly commands: CommandRegistry;
-    @inject(KeybindingRegistry) protected readonly keybindings: KeybindingRegistry;
-    @inject(MenuModelRegistry) protected readonly menus: MenuModelRegistry;
-    @inject(ScmContextKeyService) protected readonly contextKeys: ScmContextKeyService;
-    @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer;
-    @inject(ScmAvatarService) protected readonly avatarService: ScmAvatarService;
-    @inject(StorageService) protected readonly storageService: StorageService;
-    @inject(LabelProvider) protected readonly labelProvider: LabelProvider;
-    @inject(EditorManager) protected readonly editorManager: EditorManager;
-    @inject(DiffNavigatorProvider) protected readonly diffNavigatorProvider: DiffNavigatorProvider;
+    @inject(ScmCommitWidget) protected readonly commitWidget: ScmCommitWidget;
+    @inject(ScmTreeWidget) protected readonly resourceWidget: ScmTreeWidget;
+    @inject(ScmAmendWidget) protected readonly amendWidget: ScmAmendWidget;
 
-    // TODO: a hack to install DOM listeners, replace it with React, i.e. use TreeWidget instead
-    protected _scrollContainer: string;
-    protected set scrollContainer(id: string) {
-        this._scrollContainer = id + Date.now();
+    set viewMode(mode: 'tree' | 'flat') {
+        this.resourceWidget.viewMode = mode;
     }
-    protected get scrollContainer(): string {
-        return this._scrollContainer;
+    get viewMode(): 'tree' | 'flat' {
+        return this.resourceWidget.viewMode;
     }
 
-    /** don't modify DOM use React! only exposed for `focusInput` */
-    protected readonly inputRef = React.createRef<HTMLTextAreaElement>();
-
     constructor() {
         super();
         this.node.tabIndex = 0;
         this.id = ScmWidget.ID;
         this.addClass('theia-scm');
-        this.scrollContainer = ScmWidget.Styles.GROUPS_CONTAINER;
+        this.addClass('theia-scm-main-container');
     }
 
     @postConstruct()
     protected init(): void {
+        const layout = new PanelLayout();
+        this.layout = layout;
+        this.panel = new Panel({
+            layout: new PanelLayout ({
+            })
+        });
+        this.panel.node.tabIndex = -1;
+        this.panel.node.setAttribute('style', 'overflow: visible;');
```

Comment:
https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#no-styles-in-code

Reply:
I wasn't sure 'class' would work here (I did not see 'class' being used anywhere else when setting attributes on the panel node.  However it did work so I have changed it.